src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java

```diff
@@ -398,14 +401,16 @@ public void doProvision(StaplerRequest req, StaplerResponse rsp, @QueryParameter
     private void provisionAsynchronouslyNotToBlockTheRequestThread(JCloudsSlaveTemplate t) {
         Authentication auth = Jenkins.getAuthentication();
         Runnable performProvisioning = () -> {
-            try {
-                provisionSlave(t);
-            } catch (Throwable ex) {
-                LOGGER.log(Level.WARNING, "Provisioning failed", ex);
+            // Impersonate current identity inside the worker thread not to lose the owner info
+            try (ACLContext ctx = ACL.as(auth)) {
+                try {
+                    provisionSlave(t);
+                } catch (Throwable ex) {
+                    LOGGER.log(Level.WARNING, "Provisioning failed", ex);
+                }
             }
         };
-        // Impersonate current identity inside the worker thread not to lose the owner info
-        Timer.get().schedule(() -> ACL.impersonate(auth, performProvisioning), 0, TimeUnit.SECONDS);
```

Comment:
`ACL.impersonate`, while elegant, got deprecated in favor of `try-with-resources` approach.